lib/builder.dart

```diff
@@ -54,30 +54,39 @@ import 'package:coap/coap.dart';
 /// the config file to contain only those entries that override the defaults.
 /// The file can't be empty, so version must as a minimum be present.
 class $className extends DefaultCoapConfig {
-${_generateDataScript(data)}}
+${_generateDataScript(data, '')}}
 """;
 
-String _generateDataScript(YamlMap data) {
+String _generateDataScript(YamlMap data, String prefix, {firstLine = true}) {
   final buff = StringBuffer();
-  for (final k in data.keys) {
+  data.forEach((k, v) {
+    if (v is YamlMap) {
+      buff.write(_generateDataScript(v, k, firstLine: false));
+      return;
+    }
+    final variableName =
+        prefix.isEmpty ? k : prefix + k[0].toUpperCase() + k.substring(1);
+    if (!firstLine) {
+      buff.writeln('');
+    }
     buff.writeln('  @override');
-    if (data[k] is String) {
-      if (data[k] == 'true' || data[k] == 'false') {
-        buff.writeln('  bool get $k => ${data[k]};');
-        continue;
+    firstLine = false;
+    if (v is String) {
+      if ('true' == v || 'false' == v) {
```

Comment:
Tiny nitpick: I would reverse these statements, so they read `v == 'true' || v == 'false'`, putting the variable is on the left-hand side and the value it's compared to on the right-hand side.

It reads a bit easier and is in line with the rest of the code (`v == null` as opposed to `null == v`, `v is bool` as opposed to `bool is v` etc.).

Reply:
Made some adjustments that should make the code a bit more readable :)